### PR TITLE
[Testing:Developer] Refactor Pronouns Cypress Test

### PR DESCRIPTION
**Pull request by yanliw123**
refactor pronouns cypress test, using data-testids instead of using html selectors (.xxx or #xxx)

### CODE REVIEW

File: site/app/templates/EditPronounsForm.twig [hunk 1]
```diff
@@ -15,6 +15,7 @@
             <label for="user-pronouns-change">Pronouns:&emsp;</label>
             <input type="text"
                    name="user_pronouns_change"
+                   data-testid="pronounsInput"
```
Comment (IDzyre) [inline site/app/templates/EditPronounsForm.twig:18]:
```suggestion
                   data-testid="pronouns-input"
```

File: site/app/templates/EditPronounsForm.twig [hunk 2]
```diff
@@ -27,13 +28,14 @@
                     <option value="(Type Custom Here)"></option>
                 </datalist>
             </input>
-            <button aria-label="Clear pronoun input" type="button" style="background: none;border:none;" onclick="clearPronounsBox();"><i class="fas fa-trash" style="padding:0.5rem;" title="Clear pronouns field"></i></button>
+            <button aria-label="Clear pronoun input" data-testid="Clear-pronoun-input" type="button" style="background: none;border:none;" onclick="clearPronounsBox();"><i class="fas fa-trash" style="padding:0.5rem;" title="Clear pronouns field"></i></button>
```
Comment (IDzyre) [inline site/app/templates/EditPronounsForm.twig:31]:
```suggestion
            <button aria-label="Clear pronoun input" data-testid="clear-pronoun-input" type="button" style="background: none;border:none;" onclick="clearPronounsBox();"><i class="fas fa-trash" style="padding:0.5rem;" title="Clear pronouns field"></i></button>
```

File: site/app/templates/UserProfile.twig [hunk 3]
```diff
@@ -63,15 +63,15 @@
                         <span class="label">
                              {{ localize('My_Profile.Pronouns', "Pronouns" )}}:
                         </span>
-                        <button id='pronouns_val' class="icon" tabindex="0" onclick="showUpdatePronounsForm()" aria-label="Edit user pronouns" title="Edit user pronouns">
+                        <button id='pronouns_val' data-testid="pronouns_val" class="icon" tabindex="0" onclick="showUpdatePronounsForm()" aria-label="Edit user pronouns" title="Edit user pronouns">
```
Comment (IDzyre) [inline site/app/templates/UserProfile.twig:66]:
```suggestion
                        <button id='pronouns_val' data-testid="pronouns-val" class="icon" tabindex="0" onclick="showUpdatePronounsForm()" aria-label="Edit user pronouns" title="Edit user pronouns">
```

File: site/app/templates/UserProfile.twig [hunk 4]
```diff
@@ -63,15 +63,15 @@
...
                             <i class="fas fa-pencil-alt"></i> {{ user.getPronouns() }}
                         </button>
                     </div>
                     <div class="flex-row" id="pronouns-display-row">
                         <span class="label">
                             {{ localize('My_Profile.Pronouns_Forum', "Display Pronouns in Discussion Forum" )}}:
                         </span>
-                        <button id='display_pronouns_val' class="icon" tabindex="0" onclick="showUpdatePronounsForm()" aria-label="Edit user pronouns" title="Edit user pronouns">
+                        <button id='display_pronouns_val' data-testid="display_pronouns_val" class="icon" tabindex="0" onclick="showUpdatePronounsForm()" aria-label="Edit user pronouns" title="Edit user pronouns">
```
Comment (IDzyre) [inline site/app/templates/UserProfile.twig:74]:
```suggestion
                        <button id='display_pronouns_val' data-testid="display-pronouns-val" class="icon" tabindex="0" onclick="showUpdatePronounsForm()" aria-label="Edit user pronouns" title="Edit user pronouns">
```

File: site/app/templates/forum/CreatePost.twig [hunk 5]
```diff
@@ -96,14 +96,14 @@
             <a class="post-user-info key_to_click" tabindex = "0" onClick='changeName(this.parentNode, {{ post_user_info.info_name }}, {{ post_user_info.visible_user_json}}, {{ post_user_info.jscriptAnonFix}})' title="Show full user information" aria-label="Show full user information"><i class="fas fa-eye"></i></a>
         {% endif %}
 
-        <span class="last-edit">
+        <span class="last-edit" data-testid="last-edit">
             {% if post_user_info.is_OP == true and first == false %}
-			    <strong class="post_user_id" title="Original Poster">{{ visible_username }} <span style="color: var(--standard-medium-vibrant-blue);"><strong>OP</strong></span></strong>
+			    <strong class="post_user_id" data-testid="post_user_id" title="Original Poster">{{ visible_username }} <span style="color: var(--standard-medium-vibrant-blue);"><strong>OP</strong></span></strong>
```
Comment (IDzyre) [inline site/app/templates/forum/CreatePost.twig:101]:
```suggestion
			    <strong class="post_user_id" data-testid="post-user-id" title="Original Poster">{{ visible_username }} <span style="color: var(--standard-medium-vibrant-blue);"><strong>OP</strong></span></strong>
```

File: site/app/templates/forum/CreatePost.twig [hunk 6]
```diff
@@ -96,14 +96,14 @@
...
             {% else %}
-                <strong class="post_user_id">{{ visible_username }}</strong>
+                <strong class="post_user_id" data-testid="post_user_id">{{ visible_username }}</strong>
```
Comment (IDzyre) [inline site/app/templates/forum/CreatePost.twig:103]:
```suggestion
                <strong class="post_user_id" data-testid="post-user-id">{{ visible_username }}</strong>
```

File: site/app/templates/forum/CreatePost.twig [hunk 7]
```diff
@@ -96,14 +96,14 @@
...
             {% endif %}
             {% if post_user_info.pronouns and post_user_info.display_pronouns and visible_username !="Anonymous" %}
-                <strong class="post_user_pronouns">({{post_user_info.pronouns}})</strong>
+                <strong class="post_user_pronouns" data-testid="post_user_pronouns">({{post_user_info.pronouns}})</strong>
```
Comment (IDzyre) [inline site/app/templates/forum/CreatePost.twig:106]:
```suggestion
                <strong class="post_user_pronouns" data-testid="post-user-pronouns">({{post_user_info.pronouns}})</strong>
```

File: site/cypress/e2e/Cypress-UI/pronouns_change.spec.js [hunk 8]
```diff
@@ -66,73 +64,73 @@ describe('Tests cases abut changing user pronouns', () => {
...
+        cy.get('[data-testid="flex-row"]').contains('Test pronouns display').should('not.exist');
 
         // verify pronouns do not exist (post thread Anonymously) and remove thread
-        cy.contains('Test Anonymous thread, should not show pronouns').find('.post_user_id').should('contain', 'Anonymous');
-        cy.contains('Test Anonymous thread, should not show pronouns').find('.post_user_pronouns').should('not.exist');
-        cy.contains('Test Anonymous thread, should not show pronouns').find('.post_user_id').click();
-        cy.get('.post_user_id').should('contain', 'Anonymous');
-        cy.get('.post_user_pronouns').should('not.exist');
+        cy.get('[data-testid="post-action-container"]').eq(0).should('contain', 'Anonymous');
+        cy.get('[data-testid="post-action-container"]').eq(0).find('[data-testid="post_user_pornouns"]').should('not.exist');
```
Comment (IDzyre) [inline site/cypress/e2e/Cypress-UI/pronouns_change.spec.js:129]:
```suggestion
        cy.get('[data-testid="post-action-container"]').eq(0).find('[data-testid="post-user-pronouns"]').should('not.exist');
```

**Review by IDzyre — CHANGES_REQUESTED** [review]
A few nitpicky comments about hyphens instead of underscores, but a good start overall. I will do a testing sometime later today or tomorrow.

**Comment by yanliw123** [comment]
> A few nitpicky comments about hyphens instead of underscores, but a good start overall. I will do a testing sometime later today or tomorrow.

code style fixed,need a re-review

File: site/app/templates/GlobalHeader.twig [hunk 9]
```diff
@@ -326,7 +326,7 @@
                             <i class="{{ button.getPrefix() }} {{ button.getIcon() }}"></i>
                         {% endif %}
 
-                        <span class="icon-title"> {{ button.getTitle() }} </span>
+                        <span class="icon-title" data-testid="icon_{{button.getTitle()}}"> {{ button.getTitle() }} </span>
```
Comment (IDzyre) [inline site/app/templates/GlobalHeader.twig:329]: Is this being used anywhere?

**Review by IDzyre — CHANGES_REQUESTED** [review]

**Comment by IDzyre** [comment]
If you could also add a description.

**Review by IDzyre — APPROVED** [review]

**Pull request merged**